glean-core/src/common_metric_data.rs

```diff
@@ -12,6 +12,7 @@ use crate::Glean;
 ///
 /// A metric's lifetime determines when its stored data gets reset.
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[repr(i32)] // Use i32 to be compatible with our JNA definition
```

Comment:
We should confirm that this change doesn't break the database representation.

Reply:
Afaik we don't store lifetimes as such in the database.
Apart from that:

* Rust optimizes size to be as small as possible (`Lifetime` was a u8 before)
* bincode encodes enums as u32, see https://github.com/servo/bincode#details

Reply:
> Afaik we don't store lifetimes as such in the database.

Yeah -- I had to confirm my memory.  `Lifetime` isn't a key in the database -- we have separate store objects for each.

> Apart from that:
> 
>     * Rust optimizes size to be as small as possible (`Lifetime` was a u8 before)
> 
>     * bincode encodes enums as u32, see https://github.com/servo/bincode#details

Given that we don't actually store the enums in the database, I think this is ok.  Do you agree?

Reply:
Yup.